src/components/sidebar/routePathView/RoutePathViewForm.tsx

```diff
@@ -1,216 +1,246 @@
 import * as React from 'react';
+import { IRoutePath } from '~/models';
 import InputContainer from '../InputContainer';
 import { Button, Dropdown, Checkbox } from '../../controls';
 import ButtonType from '../../../enums/buttonType';
 import * as s from './routePathView.scss';
 
 interface IRoutePathViewFormState {
+    routePath: IRoutePath;
 }
 
 interface IRoutePathViewFormProps {
     isEditingDisabled: boolean;
+    routePath: IRoutePath;
+    onEdit: Function;
 }
 
 class RoutePathViewForm extends React.Component<IRoutePathViewFormProps, IRoutePathViewFormState>{
-    constructor(props: any) {
+    constructor(props: IRoutePathViewFormProps) {
         super(props);
+        this.state = {
+            routePath: this.props.routePath,
+        };
+    }
 
+    public onClick = () => {
+        // TODO
     }
 
-    // TODO
-    public onChange = () => {
+    public onChange = (property: string) => (value: string) => {
+        this.props.onEdit();
+        this.setState({
+            routePath: { ...this.state.routePath, [property]: value },
+        });
```

Comment:
I would call onChange of parent instead.

Then routePath to this form could always be given only through props (no need to have it at state)

Reason: at NewRoutePathView there is save() method that takes the edited routePath data and sends it to the backend.

So rename onEdit() -> onChange()?

Reply:
I noticed this too, when I started to make the continuation task. There I have moved the logic to the parent, I also added a comment there to consider moving the onEdit -> onChange.
https://github.com/HSLdevcom/jore-map-ui/pull/338/files#diff-7303060f0e3fa68e5bcb30f851aaa7d7R79